### DISCUSSION TITLE
【更新日志】2024/02/29 PaoPaoDNS 改进SHUFFLE功能，增加lite选项【建议升级】

### DISCUSSION THREAD

**Discussion by kkkgo** [kkkgo/PaoPaoDNS#108] [更新日志]
# 旧版本SHUFFLE功能存在问题，建议使用该功能的尽快升级

- 改进SHUFFLE洗牌功能，旧版本SHUFFLE功能会导致某些较旧的软件/库无法正确处理CNAME链导致无法解析结果/解析缓慢。在洗牌时，不对非匹配请求的记录洗牌以增加兼容性；
- SHUFFLE功能增加`lite`选项，当`SHUFFLE`设置为`lite`，会过滤掉非匹配请求类型的记录后再进行洗牌，客户端得到精简的记录回应。
以下为举例：
SHUFFLE为`no`或者`yes`时，：
![image](https://github.com/kkkgo/PaoPaoDNS/assets/4530897/32951b51-e9e2-4661-8212-01f1ee38d223)
SHUFFLE为`lite`时：
![image](https://github.com/kkkgo/PaoPaoDNS/assets/4530897/683a1a91-f1d2-4664-a329-12a7f919e09b)

注：SHUFFLE功能是对每次查询都进行洗牌输出。即使设置为no，在DNS的ttl过期后重新提供的DNS记录本身是经过unbound洗牌过的。

**Comment by fsiisx** [comment]
大佬图片挂啦

**Reply by kkkgo** [reply]
是描述不对要重新改。

**Reply by kkkgo** [reply]
改好了。
